daemon/k8s_watcher.go

```diff
@@ -1445,9 +1478,21 @@ func (d *Daemon) updateCiliumNetworkPolicyV2(ciliumV2Store cache.Store,
 		return
 	}
 
-	// Ignore updates of the spec remains unchanged.
+	// Do not add rule into policy repository if the spec remains unchanged, as
+	// policy recalculation is not needed.
 	if oldRuleCpy.SpecEquals(newRuleCpy) {
+		// If the annotations differ between the rules, but the specs are the same,
+		// just update the annotations within the CNP new rule directly, but
+		// only if the policy has already been realized for all endpoints.
+		if !oldRuleCpy.AnnotationsEquals(newRuleCpy) {
+			err := updateRuleAnnotations(newRuleCpy)
```

Comment:
This will be a blocking operation. Which will cause all policies updates / add / deletion to be blocked until this function is finish. Can't this be a controller the same way we have on line 1324 `k8sCM.UpdateController ...`?

Reply:
Sorry for the delay. Finally tackling this issue as part of the 1.2-bugfix milestone. This is a valid concern. Should this be the same controller as in `addCiliumNetworkPolicyV2` since the CNP is the same? I.e., would you recommend:

```
ctrlName := cnp.GetControllerName()
k8sCM.UpdateController(ctrlName,
    controller.ControllerParams{
        DoFunc: func() error {
            return updateRuleAnnotations(newRuleCpy)
        }
    }
)
```

This ensures that there is only one controller running for the given CNP. Also, since there is no `RunInterval` specified, it will only run once.